### DISCUSSION TITLE
How to expand IFieldUserValue.Principal

### DISCUSSION THREAD

**Discussion by LegendaryB** [pnp/pnpcore#770] [Q&A] [answered]
Hey,

how could I retrieve the Principal field of a IFieldUserValue object? It was retrieved from the IListItem.Values collection. I need it to check if this value is a Group or a User. I know that I can retrieve a ISharePointUser instance for a group and also a user via context.Web.GetUserByIdAsync. But I'm asking if there is a way to get that information without that? Or maybe getting the AadObjectId without a extra load/call to AsGraphUserAsync()?

Kind Regards

**Comment by jansenbe** [comment, marked as answer]
@LegendaryB : probably the most effective approach is to read the `IWeb` properties `SiteUsers` and `SiteGroups` once and then lookup the ids from there. Or alternatively read the "User Information List" which you can get via the `IWeb` `SiteUserInfoList` property. Key will be to read this data once and then use for all the individual list items you want to check.
